src/irmin/tree.ml

```diff
@@ -1986,32 +1984,6 @@ module Make (P : Backend.S) = struct
           false
       | Some false -> true
     in
-    let skip n =
-      match Node.cached_key n with
-      | Some k ->
-          cnt.node_mem <- cnt.node_mem + 1;
-          P.Node.mem node_t k
-      | None -> (
-          (* XXX: should we compute the actual hash here if it's not cached? *)
```

Comment:
No it should not be done because computing the hash requires exporting the children first.